sunbird_service_api_test/src/test/java/org/sunbird/integration/test/page/CreatePageTest.java

```diff
@@ -0,0 +1,132 @@
+  public static final String TEST_NAME_CREATE_PAGE_FAILURE_WITH_EXISTING_NAME =
+      "testCreatePageFailureWithExistingName";
+
+  public static final String TEST_NAME_CREATE_PAGE_SUCCESS_WITH_NAME =
+      "testCreatePageSuccessWithName";
+  public static final String TEST_NAME_CREATE_PAGE_SUCCESS_WITH_NAME_AND_ORG_ID =
+      "testCreatePageSuccessWithNameAndOrgId";
+  public static final String TEST_NAME_CREATE_PAGE_SUCCESS_WITH_PORTAL_MAP =
+      "testCreatePageSuccessWithPortalMap";
+  public static final String TEST_NAME_CREATE_PAGE_SUCCESS_WITH_APP_MAP =
+      "testCreatePageSuccessWithAppMap";
+
+  public static final String TEMPLATE_DIR = "templates/page/create";
+  public static final String ORG_CREATE_ORG_TEMPLATE_DIR = "templates/organisation/create";
+
+  private String getCreatePageUrl() {
+
+    return getLmsApiUriPath("/api/data/v1/page/create", "/v1/page/create");
+  }
+
+  @DataProvider(name = "createPageFailureDataProvider")
+  public Object[][] createPageFailureDataProvider() {
+
+    return new Object[][] {
+      new Object[] {
+        TEST_NAME_CREATE_PAGE_FAILURE_WITHOUT_ACCESS_TOKEN, false, HttpStatus.UNAUTHORIZED
+      },
+      new Object[] {TEST_NAME_CREATE_PAGE_FAILURE_WITHOUT_NAME, true, HttpStatus.BAD_REQUEST},
+      new Object[] {TEST_NAME_CREATE_PAGE_FAILURE_WITH_EXISTING_NAME, true, HttpStatus.BAD_REQUEST},
+    };
+  }
+
+  @Test(dataProvider = "createPageFailureDataProvider")
+  @CitrusParameters({"testName", "isAuthRequired", "httpStatusCode"})
+  @CitrusTest
+  public void testCreatePageFailure(
+      String testName, boolean isAuthRequired, HttpStatus httpStatusCode) {
+    getAuthToken(this, isAuthRequired);
+
+    if (testName.equalsIgnoreCase(TEST_NAME_CREATE_PAGE_FAILURE_WITH_EXISTING_NAME)) {
+      variable("pageName", PAGE_NAME);
+      beforeTestCreatePage();
+    }
+
+    performPostTest(
+        this,
+        TEMPLATE_DIR,
+        testName,
+        getCreatePageUrl(),
+        REQUEST_JSON,
+        MediaType.APPLICATION_JSON,
+        isAuthRequired,
+        httpStatusCode,
+        RESPONSE_JSON);
+  }
+
+  @DataProvider(name = "createPageSuccessDataProvider")
+  public Object[][] createPageSuccessDataProvider() {
+
+    return new Object[][] {
+      new Object[] {TEST_NAME_CREATE_PAGE_SUCCESS_WITH_NAME, true, HttpStatus.OK},
+      new Object[] {TEST_NAME_CREATE_PAGE_SUCCESS_WITH_NAME_AND_ORG_ID, true, HttpStatus.OK},
+      new Object[] {TEST_NAME_CREATE_PAGE_SUCCESS_WITH_PORTAL_MAP, true, HttpStatus.OK},
+      new Object[] {TEST_NAME_CREATE_PAGE_SUCCESS_WITH_APP_MAP, true, HttpStatus.OK},
+    };
+  }
+
+  @Test(dataProvider = "createPageSuccessDataProvider")
+  @CitrusParameters({"testName", "isAuthRequired", "httpStatusCode"})
+  @CitrusTest
+  public void testCreatePageSuccess(
+      String testName, boolean isAuthRequired, HttpStatus httpStatusCode) {
+
+    getAuthToken(this, true);
+    if (testName.equalsIgnoreCase(TEST_NAME_CREATE_PAGE_SUCCESS_WITH_NAME_AND_ORG_ID)) {
```

Comment:
Call beforeTestCreateOrg(testName) method which accepts testName as argument. Have the if check in beforeTestCreateOrg.